src/likelihoods/negativebinomial.jl

```diff
@@ -1,36 +1,169 @@
+abstract type NBParam end
+
+abstract type NBParamProb <: NBParam end
+default_invlink(::NBParamProb) = logistic
+abstract type NBParamMean <: NBParam end
+default_invlink(::NBParamMean) = exp
+
 """
-    NegativeBinomialLikelihood(l=logistic; successes::Real=1)
+    NegativeBinomialLikelihood(param::NBParam, link::Union{Function,Link})
```

Comment:
arrrgh and it's not the link, it's the _inv_link ... (why did you do that, statisticians???)

Reply:
should be changed here and everywhere else ...

Reply:
Let's keep this for another PR maybe :)

Reply:
ok, if you open the issue for it ;)

Reply:
#83